libraries/AP_Proximity/AP_Proximity_RPLidarA2.cpp

```diff
@@ -251,8 +251,13 @@ void AP_Proximity_RPLidarA2::get_readings()
     uint32_t nbytes = _uart->available();
 
     while (nbytes-- > 0) {
-
-        uint8_t c = _uart->read();
+        int16_t readChar = _uart->read();
```

Comment:
I do not fully understand what are you trying to do here. Can you please elaborate?

Reply:
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_RangeFinder/AP_RangeFinder_Benewake.cpp#L89
I detected a checksum error because I was not processing to return UART read error (return value, -1) in TF 02.
With this correspondence, in case of UART read error, we modified to get next data.
I think the same process is necessary.

UART Driver:
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_HAL_ChibiOS/UARTDriver.cpp#L501